### PR TITLE
login_call_updateUserInfos

### LINKED ISSUE #838: 🔧 [Refactor] Implement Login Call updateUserInfos on Apollo
## 🔧 Refactor ticket

Implement Login Call updateUserInfos on Apollo

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

This implements the call `updateUserInfos` on Apollo where it was implemented on the `login_server` beforehand.

Remarks:
- setting the `disabled` flag was removed
- `publisherId` is now set on login directly, not in updateUserInfos

### Issues

- fixes #838 

### Todo

- [x] Transactions for the new settings table
- [x] Change password

### CODE REVIEW

File: backend/src/graphql/resolver/UserResolver.ts [hunk 1]
```diff
@@ -455,85 +464,96 @@ export class UserResolver {
       description,
       username,
       language,
-      publisherId,
```
Comment (Elweyn) [inline backend/src/graphql/resolver/UserResolver.ts:458]: Don't we want to update the publisherId anymore?

File: backend/src/graphql/arg/UpdateUserInfosArgs.ts [hunk 2]
```diff
@@ -17,9 +17,6 @@ export default class UpdateUserInfosArgs {
   @Field({ nullable: true })
   language?: string
 
-  @Field({ nullable: true })
-  publisherId?: number
```
Comment (Elweyn) [inline backend/src/graphql/arg/UpdateUserInfosArgs.ts:21]: No publisherId update anymore?

File: backend/src/graphql/resolver/UserResolver.ts [hunk 3]
```diff
@@ -230,10 +239,10 @@ export class UserResolver {
...
     if (!user.hasElopage && publisherId) {
       user.publisherId = publisherId
-      await this.updateUserInfos(
-        { publisherId },
-        { sessionId: result.data.session_id, pubKey: result.data.user.public_hex },
-      )
+
+      const loginUser = await LoginUser.findOneOrFail({ email: userEntity.email })
+      loginUser.publisherId = publisherId
+      loginUser.save()
```
Comment (Elweyn) [inline backend/src/graphql/resolver/UserResolver.ts:245]: I am not sure if it has anything to do with the context of this PR!?
I never know when the above line work and when not so:
```suggestion
      const loginUserRepository = getCustomRepository(LoginUserRepository)
      const loginUser = await loginUserRepository.findOneOrFail({ email: userEntity.email })
      loginUser.publisherId = publisherId
      loginUserRepository.save(loginUser)
```

File: backend/src/graphql/resolver/UserResolver.ts [hunk 4]
```diff
@@ -455,85 +464,96 @@ export class UserResolver {
...
-        'User.publisher_id': publisherId || undefined,
-        'User.password': passwordNew || undefined,
-        'User.password_old': password || undefined,
-      },
-    }
-    let response: UpdateUserInfosResponse | undefined
+  ): Promise<boolean> {
     const userRepository = getCustomRepository(UserRepository)
+    const userEntity = await userRepository.findByPubkeyHex(context.pubKey)
+    const loginUser = await LoginUser.findOneOrFail({ email: userEntity.email })
```
Comment (Elweyn) [inline backend/src/graphql/resolver/UserResolver.ts:475]: Same reflexion as above never know when LoginUser.findOneOrFail or LoginUser.count works or not so I would suggest:
```suggestion
    const loginUserRepository = getCustomRepository(LoginUserRepository)
    const loginUser = await loginUserRepository.findOneOrFail({ email: userEntity.email })
```

**Review by Elweyn — CHANGES_REQUESTED** [review]
Please take the branch apollo_jwt_withdrawSessionId

File: backend/src/util/sendEMail.ts [hunk 5]
```diff
@@ -2,7 +2,12 @@ import { createTransport } from 'nodemailer'
 
 import CONFIG from '../config'
 
-export const sendEMail = async (emailDef: any): Promise<boolean> => {
+export const sendEMail = async (emailDef: {
+  from: string
+  to: string
+  subject: string
+  text: string
+}): Promise<boolean> => {
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/src/util/sendEMail.ts:10]: True, but would you be willing to turn a blind eye if we can merge this now - if not i can outsource this change.

**Review by Elweyn — APPROVED** [review]
Seems good for me! Nice work!

**Pull request merged**